tests/pious/pio/test_aggregate.py

```diff
@@ -113,3 +118,262 @@ def test_aggregate_single_file_custom_config_01():
+        assert len(result) == 1326
+
+def test_get_real_action_freqs(QJ4_c_b18_b54):
+    node_id = "r:0:c:b18:b54"
+    actions = QJ4_c_b18_b54.solver.show_children_actions(node_id)
+    sorted_actions = ag.get_sorted_actions(actions)
+    result = ag.get_real_action_freqs(QJ4_c_b18_b54, node_id, "IP",sorted_actions)
+    assert abs(result[0] - 34.11) < 0.1
+    assert abs(result[1] - 59.58) < 0.1
+    assert abs(result[2] - 2.8) < 0.1
+    assert abs(result[3] - 3.49) < 0.1
+
+def test_compute_row_matchups(QJ4_c_b18_b54):
+    ag_config = ag.AggregationConfig(global_freq=False,
+                                     evs=False,
+                                     equities=False,
+                                     action_freqs=False,
+                                     action_evs=False,
+                                     matchups=True)
+
+    result = ag.compute_row(ag_config, QJ4_c_b18_b54, 1)
+
+    # board
+    assert result[0] == "QsJs4h"
+
+    # global_freq
+    assert close_enough(result[1], 22676244.371, 0.1, False)
+
+def test_compute_row(QJ4_c_b18_b54):
+    ag_config = ag.AggregationConfig(global_freq=True,
+                                     evs=True,
+                                     equities=True,
+                                     action_freqs=True,
+                                     action_evs=True,
+                                     matchups=False)
+
+    children_actions = QJ4_c_b18_b54.available_actions()
+    sorted_actions = ag.get_sorted_actions(children_actions)
+    result = ag.compute_row(ag_config, QJ4_c_b18_b54, 1,children_actions,sorted_actions)
+
+    # board
+    assert result[0] == "QsJs4h"
+
+    # global_freq
+    assert close_enough(result[1], 0.4255, 0.1)
+
+    # evs
+    assert close_enough(result[2], 97.39, 0.1)
+    assert close_enough(result[3], 29.60, 0.1)
+
+    # eqs
+    assert close_enough(result[4], 0.5749, 0.1)
+    assert close_enough(result[5], 0.425, 0.1)
+
+    # children action_freqs
+    assert close_enough(result[6], 34.11, 0.1)  # fold
+    assert close_enough(result[7], 59.58, 0.1)  # call
+    assert close_enough(result[8], 2.8038, 0.1)  # b119
+    assert close_enough(result[9], 3.4964, 0.1)  # b280
+
+    # children action_evs (sum ev for every combo in node)
+    assert close_enough(result[10], 0, 0.1)  # fold
+    assert close_enough(result[11], 26.41, 0.1)  # call
+    assert close_enough(result[12], 1.54, 0.1)  # b119
+    assert close_enough(result[13], 1.64, 0.1)  # b280
+
+def test_aggregate_line_for_solver(QJ4_c_b18_b54):
+    ag_config = ag.AggregationConfig(equities=False,
+                                evs=False,
+                                action_freqs=False,
+                                action_evs=False,
+                                global_freq=True,
+                                matchups=True)
+
+    result = ag.aggregate_line_for_solver(QJ4_c_b18_b54.board(),
+                                          QJ4_c_b18_b54.solver,
```

Comment:
Directly use `solver` fixture dependency:

```suggestion
                                          solver,
```